tests/test_save_load.py

```diff
@@ -70,21 +70,26 @@ def test_save_load(tmp_path, model_class):
     # Check
     model.save(tmp_path / "test_save.zip")
     del model
-    model = model_class.load(str(tmp_path / "test_save.zip"), env=env)
 
-    # check if params are still the same after load
-    new_params = model.policy.state_dict()
+    # Check if the model loads as expected for every possible choice of device:
+    for device in ["auto", "cpu", "cuda"]:
```

Comment:
I noticed that the git code comparison looks quite messy. I'm elaborating about the changes I've made here to ease the review process for you:
The actual change that I made here is the added 'for' loop that goes over all possible devices, and at each iteration the device parameter is passed to the call of 'load' (line 76). At the end of each iteration I delete the model (line 92) so it can be loaded cleanly at the next iteration.
Everything else is the same as before, i.e., I've used the same test to ensure proper loading and tested with all possible values of the new argument 'device'.

Reply:
it seems that you are actually not testing that the device parameter was successfully used.
Also, you should skip the cuda device if no GPU is available

Reply:
1) You're right. I will work to improve the test.
2) What should be the expected behavior when a user uses "device='cuda'" on a machine with no GPU?
I noticed that the c'tor defaults to using the CPU in that case without notifying the user.
Anyway, I think the test should include all possible inputs while verifying that the outcome matches your expectations. Do you agree?

Reply:
I've used in my test the utils.get_device() function (which is used inside the constructor as well) to determine the device. This way, if for example, the behavior of get_device will change so that 'auto' defaults to 'cpu', the test won't break.